### ISSUE TITLE
Versions with non-digits are not interpreted correctly by UserAgentParser

### ISSUE THREAD

**Issue by NielsLeenheer** [ThaDafinser/UserAgentParser#36]
WhichBrowser adds the nickname of the OS release before the version number. So the version is "Yosemite 10.10" and not just "10.10". UserAgentParser interprets this as "0.10".

See for example:
http://useragent.mkf.solutions/?userAgent=Kodi%2F14.0+%28Macintosh%3B+Intel+Mac+OS+X+10_10_3%29+App_Bitness%2F64+Version%2F14.0-Git%3A2014-12-23-ad747d9-dirty

This also happens with many other libraries with versions like "XP" for Windows XP. It is interpreted by UserAgentParser as "0".

See for example:
http://useragent.mkf.solutions/?userAgent=Microsoft+Office%2F14.0+%28Windows+NT+5.1%3B+Microsoft+Outlook+14.0.4536%3B+Pro%3B+MSOffice+14%29

The function `getParts()` should probably skip over any non digit characters in the version string before splitting it into parts. 

Also when `setComplete()` is called, it first stores the original string as the `$complete` property, then hydrates all the parts. But the `hydrateVersionParts()` function calls `setMajor()`, `setMinor()` and `setPatch()` which in turn call `calculateComplete()` which overwrites the original `$complete` property that was initially set. This means the original string calculated by the parser is lost.

**Comment by ThaDafinser** [comment]
This seems to be a real problem which i not covered until yet.

After checking the two UA you provided, following providers are affected by this two examples
- DonatjUAParser
- SingergiBrowserDetector
- PiwikDeviceDetector
- WhichBrowser
- YzalisUAParser

Major/minor/patch are numbers, which i want to keep clean with numbers only. The question is if their are `real` version numbers which are not numeric?
E.g. alpha, beta, RC, ...?? Or dates are sometimes also used as versioning? We need here some examples.

Reason is simple: Mixing is always bad, so comparisons like `$os->getVersion()->getMajor() >= 7` wont work correctly anymore.

But OS nicknames are nice, so we should support them.

We may could/should also show it in the complete string. Or we split it to multiple methods.
`$version->getNumberComplete(); // only numeric version with . (i dont like the method name yet)`
`$version->getComplete() // nickname with numeric version?`

``` php
class Version{

    // number
    private $major;
    // number
    private $minor;
    // number
    private $patch;

    // numeric with . seperator
    private $completeNumeric

     // string e.g. XP
    private $alias;

    // string e.g. XP 5.1 | or only numeric?
    private $complete;


    // maybe a stability flag?
    private $stability;

}
```

Other things to keep in mind, if we do it in this kind of way
- support stability flag (beta, rc, alpha,...)
- ..?

**Comment by ThaDafinser** [comment]
Beta example:

```
Mozilla/5.0 (X11; U; Linux i686; xx; rv:1.9b5) Gecko/2008052519 CentOS/3.0b5-0.beta5.6.el5.centos Firefox/3.0b5
```

Alpha example:

```
Mozilla/5.0 (Android 4.3.1; rv:6.0) Gecko/20100101 Firefox/26.0prealpha
```

**Comment by NielsLeenheer** [comment]
WhichBrowser can provide the version in two ways:
- A string that is meant to be displayed on screen
- A string that is contains a version number - same specificity as the UA but reformatted

Sometimes the display string contains a prefix and the version number. Sometimes just an alias. And sometimes just the major version.

For example:
- On OS X: "Yosemite 10.10" and "10.10"
- On Windows: "XP" and "5.1"
- And for Chrome: "30" and "30.0.1599.17"

**Comment by ThaDafinser** [comment]
This should be fixed now with PR #38 
- complete is always used `as is`
- the hydration of major/minor/patch/alias was refactored that it should cover most cases
  - major/minor/patch is still number only
  - alias is populated when there is something else than number provided and not alpha/beta/...
